src/Codeception/Lib/InnerBrowser.php

```diff
@@ -72,6 +72,7 @@ public function _failed(TestInterface $test, $fail)
         
         $filename = mb_strcut($filename, 0, 244, 'utf-8') . '.fail.' . $extension;
         $this->_savePageSource($report = codecept_output_dir() . $filename);
+        $test->getMetadata()->addReport('html', $report);
```

Comment:
Quite often responses are in XML and JSON formats, does it make sense to add them as HTML reports?
Wouldn't it be better to add this report only if $extension === 'html' ?

Reply:
@Naktibalda Good thought. Hm. IIRC this (saving it non-conditionally as report 'html', which is then picked up by HTML report generator) was just the previous working logic that IMHO accidentially got lost.

Frankly I have no idea how it behaves, or in other words if it helps, when XML and/or JSON are stored as well: if the failure report is readable and/or makes failure analysis any easier. Do you know? My naive guess would be that this is also helpful for JSON and XML nonetheless, because you then have a link (that is confusingly called 'HTML snapshot') that shows the full JSON or XML response. Therefore I would just keep the fix as is, as it definitely fixes this issue and does not make it any worse with respect to before. Of course someone else can always add further improvements (later on), but this is "just" a fix (to get back the old helpful behaviour).